x/stakeibc/keeper/hooks.go

```diff
@@ -81,6 +81,31 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochN
 			// Iterate the zones and apply icaStake
 			k.IterateHostZones(ctx, icaStake)
 		}
+		// process withdrawals
+		// TODO(TEST-88): restructure this to be more efficient, we should only have to loop
+		// over host zones once
+		reinvestInterval := int64(k.GetParam(ctx, types.KeyReinvestInterval))
+		if epochNumber%reinvestInterval == 0 {
+			icaReinvest := func(index int64, zoneInfo types.HostZone) (stop bool) {
+				// Verify the delegation ICA is registered
```

Comment:
Should we also verify that the withdrawal ICA is registered (`GetWithdrawalAccount()`)? We need both to ReinvestRewards